install/kubernetes/tetragon/values.yaml

```diff
@@ -68,6 +68,11 @@ tetragon:
   #  grpc:
   #    port: 54321
 
+  # -- Use http-based livenessProbe for the tetragon container. livenessProbe has a higher priority.
+  livenessHttpProbe:
+    enabled: true
+  # host: "localhost"
```

Comment:
Maybe we could do something like:
```yaml
{{- if .Values.tetragon.livenessProbe }}
  livenessProbe:
     timeoutSeconds: 60
{{- if .Values.tetragon.livenessProbe.custom }}  
  {{- toYaml .Values.tetragon.livenessProbe.custom | nindent 4 }}
{{- else if .Values.tetragon.livenessProbe.http.enabled }}
     httpGet:
       path: /liveness
       port: 6789
{{- if .Values.tetragon.livenessProbe.http.host }}
       host: {{ .Values.tetragon.livenessProbe.http.host }}
{{- end }}
{{- else if .Values.tetragon.grpc.enabled }}
     exec:
       command:
       - tetra
       - status
       - --server-address
       - {{ .Values.tetragon.grpc.address }}
       - --retries
       - "5"
{{- end -}}
{{- end -}}
```

Reply:
Using that we can have:

-----

```yaml
livenessProbe:
  enabled: false
```
to disable liveness probe.

-----

```yaml
livenessProbe:
  enabled: true
  custom:
    grpc:
      port: 54321
```
to define a custom liveness probe.

-----

```yaml
livenessProbe:
  enabled: true
  http:
    enabled: true
```
to define an http-based liveness probe.

-----

```yaml
livenessProbe:
  enabled: true
```
to define our legacy "tetra status"-based  liveness probe (assuming `tetragon.grpc.enabled` is true).

Reply:
Maybe I'm missing something, but can we just change the default `livenessProbe` and keep everything else the same? That is, change the default values to:
```yaml
livenessProbe:
  timeoutSeconds: 60
  httpGet:
    path: /liveness
    port: 6789
```
and keep `_container_tetragon.tpl` unchanged. Then users can still use the tetra probe by setting `livenessProbe: {}`.

Reply:
Yes, this makes sense (easier compared to what I am trying to do). But what I am not able to achieve is setting  `livenessProbe: {}`. Possibly I am missing something.

First, change the existing `values.yaml` with:
```diff
@@ -64,9 +64,11 @@ tetragon:
   securityContext:
     privileged: true
   # -- Overrides the default livenessProbe for the tetragon container.
-  livenessProbe: {}
-  #  grpc:
-  #    port: 54321
+  livenessProbe:
+    timeoutSeconds: 60
+    httpGet:
+      path: "/liveness"
+      port: 6789

   # Tetragon puts processes in an LRU cache. The cache is used to find ancestors
   # for subsequently exec'ed processes.
```

Then I run:
```yaml
$ cat values.yaml
tetragon:
  livenessProbe: {}
$ helm template ./install/kubernetes/tetragon -f ./values.yaml
[...]
        livenessProbe:
          httpGet:
            path: /liveness
            port: 6789
          timeoutSeconds: 60
[...]
```
Which seems to be the same.

Reply:
We should define

```
$ cat values.yaml
tetragon:
  livenessProbe: null
```

instead of 

```
$ cat values.yaml
tetragon:
  livenessProbe: {}
```

to use the old tetra-based liveness probe. 

PR is ready for review.